scheduler/feasible.go

```diff
@@ -140,26 +140,37 @@ func NewRandomIterator(ctx Context, nodes []*structs.Node) *StaticIterator {
 type HostVolumeChecker struct {
 	ctx           Context
 	volumeReqs    []*structs.VolumeRequest
-	hostVolumeIDs []string
 	namespace     string
+	jobID         string
+	taskGroupName string
+	claims        []*structs.TaskGroupHostVolumeClaim
 }
 
 // NewHostVolumeChecker creates a HostVolumeChecker from a set of volumes
 func NewHostVolumeChecker(ctx Context) *HostVolumeChecker {
-	return &HostVolumeChecker{
-		ctx:           ctx,
-		volumeReqs:    []*structs.VolumeRequest{},
-		hostVolumeIDs: []string{},
+	hostVolumeChecker := &HostVolumeChecker{
+		ctx:        ctx,
+		claims:     []*structs.TaskGroupHostVolumeClaim{},
+		volumeReqs: []*structs.VolumeRequest{},
 	}
+
+	return hostVolumeChecker
 }
 
 // SetVolumes takes the volumes required by a task group and updates the checker.
-func (h *HostVolumeChecker) SetVolumes(
-	allocName, ns string, volumes map[string]*structs.VolumeRequest, allocHostVolumeIDs []string,
-) {
+func (h *HostVolumeChecker) SetVolumes(allocName, ns, jobID, taskGroupName string, volumes map[string]*structs.VolumeRequest) {
 	h.namespace = ns
+	h.jobID = jobID
+	h.taskGroupName = taskGroupName
 	h.volumeReqs = []*structs.VolumeRequest{}
-	h.hostVolumeIDs = allocHostVolumeIDs
+
+	storedClaims, _ := h.ctx.State().GetTaskGroupHostVolumeClaimsForTaskGroup(nil, taskGroupName)
```

Comment:
This is all claims for _any_ task group with the same name, whether or not they're for this job.

Reply:
sorry for this blunder :( fixed in eb29282